### PR TITLE
docs: Added Drag and drop files Fiddle example

### PR DESCRIPTION
**Pull request by yashints**
#### Description of Change

Refs #20442

Adds the Drag and Drop files example from `electron-api-demos` into a runnable Fiddle example.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `standard` linter passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
notes: Added the Drag and Drop files example from `electron-api-demos` into a runnable Fiddle example.

### CODE REVIEW

File: docs/fiddles/native-ui/drag-and-drop/index.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      <h3>
+        Electron supports dragging files and content out from web content into
+        the operating system's world.
+      </h3>
+
+      <p>
+        Open the
+        <a href="https://electronjs.org/docs/tutorial/native-file-drag-drop"
+          >full API documentation<span class="u-visible-to-screen-reader"
+            >(opens in new window)</span
```
Comment (erickzhao) [inline docs/fiddles/native-ui/drag-and-drop/index.html]: Due to `u-visible-to-screen-reader` not being too useful here, I'd remove the `<span>` wrapping this.

File: docs/fiddles/native-ui/drag-and-drop/index.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      <h1>Drag and drop files</h1>
+
+      <h3>
+        Electron supports dragging files and content out from web content into
+        the operating system's world.
+      </h3>
+
+      <p>
+        Open the
+        <a href="https://electronjs.org/docs/tutorial/native-file-drag-drop"
```
Comment (erickzhao) [inline docs/fiddles/native-ui/drag-and-drop/index.html]: External links in `electron-api-demos` have an event listener attached that uses Electron's `shell` API to open the link in an external browser.
```js
(link) => {
  const url = link.getAttribute('href')
  if (url.indexOf('http') === 0) {
    link.addEventListener('click', (e) => {
      e.preventDefault()
      shell.openExternal(url)
    })
  }
}
```

Without a similar listener in the Fiddle, this link opens in the same Electron window.

File: docs/fiddles/native-ui/drag-and-drop/index.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Drag and drop files</title>
+  </head>
+
+  <body>
+    <div class="section-wrapper">
+      <h1>Drag and drop files</h1>
+
```
Comment (erickzhao) [inline docs/fiddles/native-ui/drag-and-drop/index.html]: Could you add the platform and process information below the title?

File: docs/fiddles/native-ui/drag-and-drop/main.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  if (mainWindow === null) {
+    createWindow()
+  }
+})
+
+ipcMain.on('ondragstart', (event, filepath) => {
+  const iconName = 'codeIcon.png'
+  event.sender.startDrag({
+    file: filepath,
+    icon: path.join(__dirname, iconName)
```
[the remark this answers is not in the document]
Comment (yashints) [reply docs/fiddles/native-ui/drag-and-drop/main.js]: Done

File: docs/fiddles/native-ui/drag-and-drop/index.html [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        Electron supports dragging files and content out from web content into
+        the operating system's world.
+      </h3>
+
+      <p>
+        Open the
+        <a href="https://electronjs.org/docs/tutorial/native-file-drag-drop"
+          >full API documentation<span class="u-visible-to-screen-reader"
+            >(opens in new window)</span
+          ></a
```
[the remark this answers is not in the document]
Comment (yashints) [reply docs/fiddles/native-ui/drag-and-drop/index.html]: done

File: docs/fiddles/native-ui/drag-and-drop/index.html (hunk 1 above)
Comment (yashints) [reply docs/fiddles/native-ui/drag-and-drop/index.html]: done

File: docs/fiddles/native-ui/drag-and-drop/index.html [hunk 6]
```diff
@@ -0,0 +1,78 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Drag and drop files</title>
+  </head>
+
+  <body>
+    <div class="section-wrapper">
```
[the remark this answers is not in the document]
Comment (yashints) [reply docs/fiddles/native-ui/drag-and-drop/index.html]: done

File: docs/fiddles/native-ui/drag-and-drop/index.html (hunk 2 above)
Comment (yashints) [reply docs/fiddles/native-ui/drag-and-drop/index.html]: added to renderer, is there a way to add this globally?

File: docs/fiddles/native-ui/drag-and-drop/index.html (hunk 3 above)
Comment (yashints) [reply docs/fiddles/native-ui/drag-and-drop/index.html]: added

**Pull request merged**